### PR TITLE
Add basics documentation - split Config and Plugin docs

### PR DESCRIPTION
**Pull request by JoranAngevaare**
**What is the problem / what does the code in this PR do**
In this PR:
 - we add basics documentation https://strax--691.org.readthedocs.build/en/691/basics/overview.html
 - Split configs and plugins documentation (https://strax--691.org.readthedocs.build/en/691/advanced/config.html and https://strax--691.org.readthedocs.build/en/691/advanced/plugin_dev.html, which used to be one huge page)
 - Add documentation for #689 in the plugin documentation https://strax--691.org.readthedocs.build/en/691/advanced/plugin_dev.html#strax-plugin
 - Remove some old files not used anymore.
 - Fix a few RST issues on https://strax--691.org.readthedocs.build/en/691/advanced/superrun.html

### CODE REVIEW

File: docs/source/basics/overview.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+Plugins
+-------------
+To a context several classes are registered that are responsible for processing/generating the data. In the example-figure above, we have three plugins:
+ - PulseProcessing
+ - PeakProcessing
+ - EventProcessing
+
+Let's start by adding the first plugin (``PulseProcessing``):
+.. code-block:: python
```
Comment (WenzDaniel) [inline docs/source/basics/overview.rst:34]: This is somehow not recognized at least it is written in the html too.

**Review by WenzDaniel — APPROVED** [review]
Thanks a lot I just found one small thing. Maybe we should ask for some feedback in the young channel? Not sure if I am qualified to evaluate if we need to be a bit more explicit here and there. I also was wondering if you want to add a few words about your nice plugin caching in the context chapter.

File: docs/source/basics/overview.rst (hunk 1 above)
Comment (JoranAngevaare) [reply docs/source/basics/overview.rst:34]: Thanks a lot, let me fix it!

**Comment by JoranAngevaare** [comment]
> Thanks a lot I just found one small thing. Maybe we should ask for some feedback in the young channel? 

Great idea! 


> Not sure if I am qualified to evaluate if we need to be a bit more explicit here and there. I also was wondering if you want to add a few words about your nice plugin caching in the context chapter.

Nice idea, but I guess that is ~ advanced level.

**Pull request merged**